lib/middleware/check-progress.js

```diff
@@ -17,8 +18,7 @@ module.exports = function checkProgress(route, controller, steps, start) {
     var invalidatePath = function (req, res) {
         var potentialPaths = _.pluck(controller.options.forks, 'target')
             .concat(controller.options.next);
-
-        var nextStep = controller.getNextStep(req, res).replace(/\/edit$/, '');
+        var nextStep = FormController.prototype.getForkTarget.apply(controller, arguments);
```

Comment:
> If I have a custom controller and I decide, for whatever crazy reason, to override the getForkTarget method (maybe I've invented a new style of forking, who knows) then I would expect my method to be used here, and presumably my method would need to be used here.

@daniel-ac-martin makes a good point - if we want to be overriding the default behaviour then this middleware should probably use it.  Lets go with that. 

We won't need proxyquire anymore but I still believe it is the _right_ way to stub dependencies 😉